### ISSUE TITLE
Version info will be missing if change offline package

### ISSUE THREAD

**Issue by lucklove** [pingcap/tiup#594]
After change from `v4.0.1` -> `v4.0.2`:
```
# export TIUP_MIRRORS=/data/tidb/tidb-ansible-dir/tiup_binary/tidb-community-server-v4.0.2-linux-amd64
# tiup cluster
WARNING: both mirror config(/data/tidb/tidb-ansible-dir/tiup_binary/tidb-community-server-v4.0.1-linux-amd64)
and TIUP_MIRRORS(/data/tidb/tidb-ansible-dir/tiup_binary/tidb-community-server-v4.0.2-linux-amd64) have been set.
Setting mirror to TIUP_MIRRORS(/data/tidb/tidb-ansible-dir/tiup_binary/tidb-community-server-v4.0.2-linux-amd64)
Failed to start component `cluster`
Error: no version: v1.0.6
```

**Comment by lucklove** [comment]
This is because we will run the installed last stable version if the user doesn't specific a version for a component(`cluster`). And we install v1.0.6, so we try to find v1.0.6 in offline package `v4.0.2` where it doesn't exists.

**Comment by lucklove** [comment]
TODO:
- [ ] Clone the whole component manifest instead of only the target version
- [ ] Yank all versions in the manifest except target version
- [ ] Deal with the yanked versions in the list command and last stable function

**Comment by AstroProfundis** [comment]
Fixed in #602, #635
